### ISSUE TITLE
Adding a but more dynamic to opfor strength level

### ISSUE THREAD

**Issue by SondreBakken** [Wyqer/kp_liberation#329]
This suggestion plays on two aspects, adding a bit more control around opfor strength, and adding back a mechanic from the original Liberation in a new form.

In the old liberation you could get resources by blowing up enemy vehicles. This was removed, because it didn't really fit well with the new resource system. But another thing that was removed then as well was the incentive to go on bombing runs to target armored targets, or use guerrilla tactics to take out some tanks without capturing sectors, or behind-the-lines ops to laser-mark armored targets for artillery.

So my suggestion goes like this; Taking out vehicles/armors lowers opfor strength. Currently you can only do this by doing the special missions. This would also play well into the logic that taking out enemy vehicles/tanks/helis etc lowers enemy strength.

Since lowering strength is now easier it could be raised a bit more when capturing a sector while a attack on the fob would not lower it automatically (but it would be lowered upon defeat in line with the size of the attack because you would take out a lot of the enemy vehicles in doing so)

This easy change would open up a lot of new game play elements and strategies

And additional suggestion would be that opfor strength would raise a certain amount every X minutes instead of being raised on taking sectors (rather it could be lowered by it), that way you would need to actively work to keep it down buy blowing up opfor vehicles and doing secondary missions (and capturing sectors).

**Comment by SondreBakken** [comment]
@dmorchard Yes, I guess the difficulty parameter would adjust at which rate the opfor strength increases :)

Some testing and balancing on the different factors that lowers/heightens opfor strength would ofc be necessary

**Comment by Wyqer** [comment]
A kind of this was planned for 0.96, but was postponed to 0.97.
It was written in the "planned features for 0.96", but I can't find it anymore, as it was also removed after release of 0.96. I should've keep the thoughts, but I'll have a look in my local files, if it's still somewhere.

**Comment by Wyqer** [comment]
Just have it still in german. Maybe google could help for an understanding.

```
Feindaufmerksamkeit und Stärke
•	Teilen der Aufmerksamkeit in zwei Werte -> Feindstärke und Aufmerksamkeit.
•	Stärke wird anzeigen, wie viel Kriegsgerät und Männer der Feind in die Schlacht etc. schicken kann.
•	Aufmerksamkeit wird anzeigen, in wie fern der Feind Deinen Fortschritt und Deine Handlungen als Bedrohung ansieht und ob er auf Deine Aktionen reagieren soll.
•	Stärke sollte gesenkt werden, in dem man Basen einnimmt oder den Feind vom Nachschub abschneidet. Sie sollte mit der Zeit erhöht werden als eine Art „Der Feind kann von dort immer noch Nachschub anfordern“ oder so ähnlich.
•	Aufmerksamkeit sollte im Grunde so bleiben, wie es zurzeit ist, bezüglich des Ansteigens und Absinkens.
•	Also kurz gesagt wird die Aufmerksamkeit anzeigen, wie oft der Feind Aktionen gegen die Spieler durchführt und Stärke wird anzeigen, wie viel Fahrzeuge und Männer ihm für jede Aktion zur Verfügung stehen.
Feindverhalten und Fähigkeiten
•	Lage der FOBs sollte für den Feind „unbekannt“ sein, nachdem sie gebaut wurden.
•	Der Feind sollte eine Art von Aufklärungsteam entsenden, oder von Zivilisten über Blufor FOBs und/oder Fähigkeiten (Anzahl der Ressourcen, Anzahl der Flugkontrollgebäude, etc.) informiert werden.
•	Kampfgruppen sollten nur bei den Feindbasen spawnen und nicht „irgendwo im offenen Gelände“. Ich denke, dass sich daraus ein strategischeres Game Play ergeben könnte, da man so planen kann, wie man Feindverstärkung oder Kampfgruppen durch Hinterhalte an möglichen Patrollienrouten etc., abschneiden kann. 
•	Wenn man Funktürme angreift, sollte der Feind über die Anwesenheit Bescheid wissen (wie die Spieler es wissen, nachdem sie einen Funkturm übernommen haben) und angemessen auf die Annäherung reagieren.
•	Die Angriffsziele des Feindes sollten sich aus der Anzahl der Ressourcen und der Entfernung zu Opfor Sektoren ergeben. Im Falle von FOBs sollten noch die Anzahl der Fahrzeuge und „Flugkontrollgebäude“ hinzukommen.
•	Es sollten keine „endlosen“ Verstärkungen für den Feind geben. Weiter noch sollte er eine limitierte Anzahl von Männern, leichten Fahrzeugen, Helikoptern, etc. haben, abhängig seiner Stärke und das Limit wird z.B. alle 3 Stunden zurückgesetzt.
•	Vielleicht eine Art von einer „hochrangigen“ Feindinstallation am Flughafen oder Schiffshafen, welche die Zeit in der das Limit zurückgesetzt wird, die Fähigkeiten oder die Stärke des Feindes in irgendeiner Art beeinflusst.
•	Mehr umherstreifende oder patrollierende Feinde im Feindterritorium, abhängig von ihrer Aufmerksamkeit.
•	Wenn mit dem Logistiksystem gespielt wird, wird es die Chance geben, dass der Feind einen Hinterhalt auf diese Konvois durchführt, abhängig von der Entfernung zu seinem Territorium, was eine Aufgabe auf Zeit für die Spieler nach sich zieht, um dies zu verhindern.
•	Möglichkeit von Mörserstellungen. Sie würden nicht wie beim normalen ArmA-Verhalten feuern (ich schätze, das wäre zu OP), aber durch ein Script.
•	Der Luftraum des Feindes sollte nicht mehr so sicher sein, wie es im Moment ist.  Denke immer noch über eine geeignete Flugabwehrtechnik innerhalb des Feindflugraumes nach, die es gefährlicher machen würde, aber nicht unmöglich für Annäherungen „hinter feindlichen Linien“. 
```

**Comment by SondreBakken** [comment]
> Enemy attention and strength
> • Divide Attention into Two Values ​​-> Enemy Strength and Attention.
> • Strength will show how much war equipment and men the enemy can send into battle etc.
> • Attention will indicate how far the enemy sees your progress and your actions as a threat and whether he should respond to your actions.

I don't know if this is necessary unless directly tied into some specific gameplay elements. Simplicity is often good when it comes to games (easy to learn, hard to master). Also missing how these things will work gameplay wise.

> • Strength should be lowered by taking bases or cutting off the enemy from supplies. It should be increased over time as a kind of "The enemy can still request replenishment from there" or something like that.

Similar to my suggestion. I like this. It adds realism :)

> • Attention should basically remain as it is, in terms of rising and falling.
> So, in a nutshell, attention will show how often the enemy performs actions against the players, and strength will indicate how many vehicles and men are available to them for each action.

So sort of aggresivity works now. Strength -> how strong attacks are and attention -> how often they attack. But what affects attention? Attacks from you could highten attention, and then it could maybe get lowered by time. So attacking small villages covered by enemy radiotowers hightens attention X while small villages not covered hightens attention a little less.
Big cities and bases hightens it more, but less if not covered by a radiotower. Taking a base lowers strength somewhat too because its a support point. Also there is more armored vehicles there so blowing up that would lower it even more.

> Enemy behavior and abilities
> • Location of the FOBs should be "unknown" to the enemy after they were built.

Agreed. Maybe you can get a notification when a FOB is discovered and some icon next to their name on the map so you know which ones are discovered. (Maybe you need to interrogate prisoners to have this information)

> • The enemy should send some type of reconnaissance team, or be informed by civilians about Blufor FOBs and / or capabilities (number of resources, number of air traffic control buildings, etc.).

Good point, but unless the player somehow can see this happening and affect it I dont think this is necessary. It adds complexity. Maybe a FOB would be discovered once seen by a passing patrol etc because then having radiotowers and attacking enemy that comes into the area before they can discover you base would be a thing, and that would be kind of cool. Maybe having a radiotower would give you the notification: "An enemy patrol is in the area of one of your undiscovered FOBS"

> • Combat groups should only spawn at the enemy bases and not "somewhere in open terrain". I think that could result in a more strategic game play, since you can plan how to cut off enemy reinforcements or combat groups by ambushing on possible patrol routes, etc.

I like this one a lot. This would add more planning to an attack in that you can look at roads and decide which ones to add mines to etc to stop reinforcements :) Also taking out a base would make it easer to capture nearby sectors afterwards

> • When attacking radio towers, the enemy should know about the presence (as the players know after they have taken over a radio tower) and react appropriately to the approach.

I like this one. Another one is more randomization around defensive forces surrounding a tower. Maybe its a tank, a offroad, a group of people, or maybe even a heli landed there.

> • The enemy's targets should be based on the number of resources and the distance to Opfor sectors. In the case of FOBs, the number of vehicles and "flight control buildings" should be added.

Not sure if I understand this one

> • There should be no "endless" reinforcements for the enemy. Furthermore, he should have a limited number of men, light vehicles, helicopters, etc., depending on his strength and the limit is e.g. reset every 3 hours.

Based on my suggestion every kind of enemy vehicle typed is tied to a strength point (or percentage).
So blowing up a jet lowers enemy strength by 3%, a heli 2%, a tank 2% and other armored and non armoured vehicles 1%. (Just an example).

Strength is increased over time, and not lowered unless you blow up stuff, capture enemy fobs or do certain sidemissions. 

This basically means that percentage strength is like a pool of enemy vehicles. So when an attack is triggered (this could be triggered by "attention" at a certain level) the strength of the attack would be translatable into the current percentage. So if strength is at 10% an attack could have 2 tanks, a heli, and and two vehicles. Destroying these would put strengt to 0%. Ofc an attack don't have to be "full" strength. At higher strength like 40%+ it could launch attacks at equal 15% strength. Kinda bad explanation but hope you get it.

Additional rules could be applied. Like strength has to be above 30% for jets to spawn, so opfor at 3% strength don't just send one jet! 

> • More roaming or patrolling enemies in the enemy territory, depending on their attention.

I like this one. It should also depend on closeness of enemy sectors. So you dont run into enemy patrols in areas surrounded by taken sectors

> • Possibility of mortar positions. They would not fire like normal ArmA behavior (I guess that would be too OP) but through a script.

I played some time with the Fire for effect mod in Liberation and it is awesome. It fires at whatever units spot, and takes into account speed and direction etc. It fires two rounds, which are very inaccurate, then spends some time adjust, and fires a more accurate round. This is really cool because you get the indication that someone has seen you / or your fob and that you have to take them out before the more accurate barrage hits / or get to cover

> • The enemy's airspace should not be as secure as it is at the moment. Still think about a suitable anti-aircraft technology within the enemy space, which would make it more dangerous, but not impossible for approaches "behind enemy lines".

I think it would work just spawn some static AA as defenders

**Comment by Wyqer** [comment]
Oh nice that you made a small "retranslation", or did you found the old english text?
Thanks for sharing it here, so everyone can have a look at "the current state of thoughts". :)

Here is a missunderstanding I guess:

>>  Attention should basically remain as it is, in terms of rising and falling.

> But what affects attention?

Attention = Aggressivity. Strength will "extend" the current Aggressivity. But the Aggressivity/Attention should stay in it's behaviour concerning how it rises and lowers.
https://github.com/Wyqer/kp_liberation/blob/master/Missionframework/scripts/server/remotecall/sector_liberated_remote_call.sqf#L3



>>The enemy's targets should be based on the number of resources and the distance to Opfor sectors. In the case of FOBs, the number of vehicles and "flight control buildings" should be added.

>>Not sure if I understand this one

Blufor sectors and FOBs should have something like a "value of worth" (internal). If they have massive resources stockiled and threee facilities or (in case of a FOB) many air vehicle slot buildings they should have a bigger "score" for the decision of the enemy which one could be attacked. Also the distance to the opfor territory should come into that decision to keep it reasonable.

**Comment by SondreBakken** [comment]
Ah, that makes sense. Then it would be preferable to place bigger bases with more resources and stuff further inside your controlled area of the map. Would also mean that moving resources from FOBs at the frontline to your "mainbase" would be reasonable.

**Comment by Takahashi-Yuuki** [comment]
I disagree with destroying enemy vehicles/troops reducing strength. My group played an old version of Liberation and we completely destroyed the enemy army just by sitting in a defensive position. The rest of the campaign after that point was almost literally just walking through the points.

Player should have to actually go out to reduce enemy army strength. That means doing missions or capturing points, etc. The FOB hunting works just fine in this regard as far as i'm concerned. Althought some disincentive to just shelling them into oblivion should exist. Perhaps if you had to go and retrieve some object from the base. Battle plans or some such thing.

**Comment by SondreBakken** [comment]
@Takahashi-Yuuki Thats a question of balancing. It does not have to be easy even though this feature would be introduced. It could even become harder if you introduce the 'increase strength per minute' to simulate opfor activily preparing to fight you.
